src/url-params.ts

```diff
@@ -175,11 +175,24 @@ export class AnalyticsUrlParams {
 	/**
 	 * @return full query parameter string that can be appended to URLs
 	 */
-	getQueryString(destinationUrl?: URL, currentUrl?: URL): string {
+	getQueryString(destinationUrl?: URL | string, currentUrl?: URL): string {
+		// we first check if destionationUrl is a relative URL string. If it is, we exit.
+		const relativeRegex = /^(?!www\.|(?:http|ftp)s?:\/\/|[A-Za-z]:\\|\/\/).*/g;
```

Comment:
So this regex catches strings like `hub.balena.io` or `balena.io`. Do we want that? Shall we exclude the strings with domain names as well?

Reply:
@gelbal  the thing is that an href thats only 'hub.balena.io' or 'hub.balena.io' shoudnt be possible since it wont work. If you use that as an `href` the HTML would send you to www.domain.com/hub.balena.io . I think that should be always be avoided in practice right?

Reply:
Aha. Good to know. Then this regex seems like working fine.